dask/array/reblock.py

```diff
@@ -0,0 +1,201 @@
+from __future__ import absolute_import, division, print_function
+"""
+The reblock module defines:
+    intersect_blockdims: a function for
+        converting blockdims to new dimensions
+    reblock: a function to convert the blocks
+        of an existing dask array to new blockdims or blockshape
+"""
+
+from itertools import count, product, chain
+from operator import getitem, add
+import numpy as np
+from toolz import merge, accumulate
+from dask.array.core import rec_concatenate, Array
+from dask.array.core import blockdims_from_blockshape
+
+
+reblock_names  = ('reblock-%d' % i for i in count(1))
+
+
+def cumdims_label(blockdims, const):
+    """ Interal utility for cumulative sum with label.
+
+    >>> cumdims_label(((5, 3, 3), (2, 2, 1)), 'n')  # doctest: +NORMALIZE_WHITESPACE
+    [(('n', 0), ('n', 5), ('n', 8), ('n', 11)),
+     (('n', 0), ('n', 2), ('n', 4), ('n', 5))]
+    """
+    return [tuple(zip((const,) * (1 + len(bds)),
+                      list(accumulate(add, (0,) + bds))))
+              for bds in blockdims ]
+
+
+def _breakpoints(cumold, cumnew):
+    """
+
+    >>> new = cumdims_label(((5, 3, 3), (2, 2, 1)), 'n')
+    >>> old = cumdims_label(((2, 2, 1), (5,)), 'o')
```

Comment:
@PeterDSteinberg I was confused about this test.  The two first blockdims tuples seem to have different sums (11 or 5).  What was the intent here?  I'm happy to make the change, just want to know what it should be changed to.